src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -473,7 +473,7 @@ Player* ScriptedAI::SelectTargetFromPlayerList(float maxdist, uint32 excludeAura
     std::vector<Player*> tList;
     for(Map::PlayerList::const_iterator itr = pList.begin(); itr != pList.end(); ++itr)
     {
-        if (me->GetDistance(itr->GetSource()) > maxdist || !itr->GetSource()->IsAlive() || itr->GetSource()->IsGameMaster())
+        if (!me->IsWithinDistInMap(itr->GetSource(), maxdist) || !itr->GetSource()->IsAlive() || itr->GetSource()->IsGameMaster())
```

Comment:
Will this break some other code?

Reply:
`GetDistance` checks only distance. `IsWithinDistInMap` checks distance, same map and same phase. I looked at other calls of this function and this change should not break anything. It ensures that the selected targets in the same phase as source. This function is used mainly to select victim to be attacked by source, so this is just improvement.

Reply:
Will it be safer to do so?
![image](https://user-images.githubusercontent.com/18535853/111301301-02c66800-868d-11eb-914a-debe51b1a27a.png)

Reply:
OK